tools/releaser/main.go

```diff
@@ -0,0 +1,270 @@
+/* Copyright 2023 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// releaser is a tool for managing part of the process to release a new version of gazelle.
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"github.com/bazelbuild/bazel-gazelle/rule"
+	bzl "github.com/bazelbuild/buildtools/build"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path"
+	"strings"
+)
+
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer cancel()
+	if err := run(ctx, os.Stderr); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func run(ctx context.Context, stderr *os.File) error {
+	var (
+		verbose bool
+	)
+
+	flag.BoolVar(&verbose, "verbose", false, "increase verbosity")
+	flag.BoolVar(&verbose, "v", false, "increase verbosity (shorthand)")
+	flag.Usage = func() {
+		fmt.Fprint(flag.CommandLine.Output(), `usage: bazel run //tools/releaser
+
+This utility is intended to handle many of the steps to release a new version.
+
+`)
+		flag.PrintDefaults()
+	}
+
+	flag.Parse()
+
+	workspacePath := path.Join(os.Getenv("BUILD_WORKSPACE_DIRECTORY"), "WORKSPACE")
+	depsPath := path.Join(os.Getenv("BUILD_WORKSPACE_DIRECTORY"), "deps.bzl")
+	_tmpBzl := "tmp.bzl"
+	tmpBzlPath := path.Join(os.Getenv("BUILD_WORKSPACE_DIRECTORY"), _tmpBzl)
```

Comment:
Ah I assumed for some reason this file needed to stay within the local workspace but looks like not.

Reply:
Actually I do not believe this is possible because the destination for to_macro must be local - https://github.com/bazelbuild/bazel-gazelle/blob/master/cmd/gazelle/update-repos.go#L229 and cannot back up directories https://github.com/bazelbuild/bazel-gazelle/blob/master/cmd/gazelle/update-repos.go#L66

Reply:
Oh, yes, you are right. So just address the `defer os.Remove(tmpBzl.Name())` part then. Although you can call `os.CreateTemp(os.Getenv("BUILD_WORKSPACE_DIRECTORY"), "tmp.*.bzl")`, it makes less sense if you are not using the system temp dir.

Reply:
I also don't think I can rely on `defer os.Remove` because I need that file to be gone before we do the final gazelle run. Let me see what I can do to be a little safer on errors though

Reply:
I see. If there is too much trouble, don't worry about that.